lib/Data/Decimate.pm

```diff
@@ -0,0 +1,204 @@
+package Data::Decimate;
+
+use strict;
+use warnings;
+
+use 5.010;
+
+use Exporter qw/import/;
+
+our @EXPORT_OK = qw(decimate);
+
+=head1 NAME
+
+Data::Decimate - A module that allows to decimate a data feed. 
+
+=head1 VERSION
+
+Version 0.01
+
+=head1 SYNOPSIS
+
+  use Data::Decimate qw(decimate);
+
+  my @data = [
+        {epoch  => 1479203101,
+        ...},
+        {epoch  => 1479203102,
+        ...},
+        {epoch  => 1479203103,
+        ...},
+        ...
+        {epoch  => 1479203114,
+        ...},
+        {epoch  => 1479203117,
+        ...},
+        {epoch  => 1479203118,
+        ...},
+        ...
+  ];
+
+  my $output = Data::Decimate::decimate(15, \@data);
+
+  #epoch=1479203114 , decimate_epoch=1479203115
+  print $output->[0]->{epoch};
+  print $output->[0]->{decimate_epoch};
+
+=head1 DESCRIPTION
+
+A module that allows you to resample a data feed
+
+=cut
+
+our $VERSION = '0.01';
+
+=head1 SUBROUTINES/METHODS
+=cut
+
+sub _fill_missing_data {
+    my ($interval, $decimate_data) = @_;
+
+    my @sorted_data = sort { $a <=> $b } keys %$decimate_data;
+    my $first_key   = $sorted_data[0];
+    my $last_key    = $sorted_data[-1];
+
+    if (scalar(@sorted_data)) {
+        for (my $i = $first_key; $i <= $last_key; $i = $i + $interval) {
+            my $data = $decimate_data->{$i};
+
+            if (not $data) {
+                my $data     = $decimate_data->{$i - $interval};
+                my %to_store = %$data;
+                $to_store{decimate_epoch} = $i;
+                $to_store{count}          = 0;
+                $decimate_data->{$i}      = \%to_store;
+            }
+        }
+    }
+
+    return $decimate_data;
+}
+
+=head2 decimate
+
+Decimate a given data based on sampling frequency.
+
+=cut
+
+sub decimate {
+    my ($interval, $data) = @_;
+
+    if (not defined $interval or not defined $data) {
+        die "interval and data are required parameters.";
+    }
+
+    my $counter             = 0;
+    my $prev_decimate_epoch = 0;
+    my %decimate_data;
+
+    if ($data) {
+        %decimate_data = map {
+            my $decimate_epoch = ($_->{epoch} % $interval) == 0 ? $_->{epoch} : $_->{epoch} - ($_->{epoch} % $interval) + $interval;
+            $counter = ($decimate_epoch == $prev_decimate_epoch) ? $counter + 1 : 1;
+            $_->{count}          = $counter;
+            $_->{decimate_epoch} = $decimate_epoch;
+            $prev_decimate_epoch = $decimate_epoch;
+            ($decimate_epoch) => $_
```

Comment:
`()` isn't needed here.

And you changed  the value of $data in this block. I think it is better no side-effect here.

I perfer to use `for` in such a long block.